tidy3d/components/simulation.py

```diff
@@ -49,9 +49,12 @@
 MAX_CELLS_TIMES_STEPS = 1e17
 MAX_MONITOR_DATA_SIZE_BYTES = 10e9
 
-# number of grid cells * structures at which we star warning about slow Simulation.epsilon()
+# number of grid cells at which we warh about slow Simulation.epsilon()
 NUM_CELLS_WARN_EPSILON = 100_000_000
+# number of structures at which we warh about slow Simulation.epsilon()
 NUM_STRUCTURES_WARN_EPSILON = 10_000
+# maximum number of structures for which we warn that grid size is too small
+MAX_STRUCTURES_WARN_GRID_SIZE = 50
```

Comment:
might even make this lower.. like 5, or just warn on the first one and say "and perhaps others" and return.